### PR TITLE
Adds PipelineRun message

### PR DESCRIPTION
**Pull request by XinruZhang**
# Changes
1. Creates pipelinerun.proto file to store PipelineRun message
2. Split out ObjectMeta and Condition messages to common.proto file
   to avoid repeat definitions

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: results/proto/pipelinerun.proto [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import "common.proto";
+import "taskrun.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "./proto";
+
+message PipelineRun {
+    string api_version = 1;
+	string kind = 2; // TODO: must always be TaskRun?
```
Comment (wlynch) [inline results/proto/pipelinerun.proto:14]: Fix indentation (here and other places in the file)

Comment (wlynch) [inline results/proto/pipelinerun.proto:14]: Can probably remove the TODO (or switch to PipelineRun to reflect the right type)

File: results/proto/pipelinerun.proto [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	ObjectMeta metadata = 3;
+
+	PipelineRunSpec spec = 4;
+	PipelineRunStatus status = 5;
+}
+
+message PipelineRunSpec {
+    PipelineSpec task_spec = 5;
```
Comment (wlynch) [inline results/proto/pipelinerun.proto:23]: Any reason why these proto numbers don't start at 1?

File: results/proto/pipelinerun.proto [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	google.protobuf.Timestamp start_time = 4;
+    google.protobuf.Timestamp completion_time = 5;
+    
+    // The type of this field in go struct is map[string] *PipelineRunTaskRunStatus
+    map<string, PipelineRunTaskRunStatus> taskruns = 6;
+    PipelineSpec pipeline_spec = 7;
+}
+
+message PipelineRunTaskRunStatus {
+    string pipeline_task_nam = 1;
```
Comment (wlynch) [inline results/proto/pipelinerun.proto:75]:
```suggestion
    string pipeline_task_name = 1;
```

We should probably add some sort of test to make sure this converts correctly. With TaskRuns there were a few cases where the Go struct name was not in the default format protobuf expected, causing issues when converting.

**Review by wlynch — CHANGES_REQUESTED** [review]

File: results/pkg/watcher/reconciler/controller_test.go [hunk 4]
```diff
@@ -132,6 +132,7 @@ func TestReconcile(t *testing.T) {
 		tr.UID = "234435"
 		asset.Clients.Pipeline.TektonV1beta1().TaskRuns(taskRun.Namespace).Update(tr)
 		updatetr, err := reconcileTaskRun(ctx, asset, tr)
+		updatetr.ResourceVersion = tr.ResourceVersion
```
Comment (wlynch) [inline results/pkg/watcher/reconciler/controller_test.go:135]: This seems unrelated to the rest of the proto changes. Why is this necessary?

File: results/proto/pipelinerun.proto [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "./proto";
+
+message PipelineRun {
+  string api_version = 1;
+  string kind = 2;
+
+	ObjectMeta metadata = 3;
```
Comment (wlynch) [inline results/proto/pipelinerun.proto:16]: Fix indenting.

**Review by wlynch — CHANGES_REQUESTED** [review]
Mostly lgtm! Few minor comments.

File: results/pkg/watcher/reconciler/controller_test.go (hunk 4 above)
Comment (XinruZhang) [reply results/pkg/watcher/reconciler/controller_test.go:135]: That's because the new version of k8s will change objects' ResourceVersion after they are updated, so we need to equal them manually before comparing them. See [concurrency-control-and-consistency](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency). Should we also make sure the ResourceVersion field is changed?

**Review by wlynch — APPROVED** [review]
/lgtm
